analyses/cell-type-ewings/exploratory_analysis/08-merged-celltypes.Rmd

```diff
@@ -0,0 +1,742 @@
+      final_annotation == "tumor EWS-high" & proliferative_mean > 0,
+      "tumor EWS-high proliferative",
+      final_annotation
+    ), 
+    final_lumped = final_annotation |>
+      forcats::fct_lump_n(10, other_level = "All remaining cell types", ties.method = "first") |>
+      forcats::fct_infreq() |>
+      forcats::fct_relevel(cell_type_order)
+  )
+
+# first get tally for how many libraries each cell type is present in 
+total_libraries <- all_info_df |> 
+  dplyr::select(final_annotation, library_id) |> 
+  unique() |> # get rid of duplicate cells
+  dplyr::group_by(final_annotation) |> 
+  dplyr::summarize(
+    total_libraries = length(library_id)) |> 
+  dplyr::arrange(desc(total_libraries))
+
+# print out both total cells and total libraries 
+all_info_df |> 
+  dplyr::count(final_annotation, name = "total_cells") |>
+  dplyr::left_join(total_libraries, by = "final_annotation") |> 
+  dplyr::arrange(desc(total_cells))
+```
+
+
+For plotting purposes, we will only show the top 10 cell types with muscle cell being the last cell type. 
+It looks like all other cell types are only present in 1-3 libraries. 
+For any cells where there's only 1 cell we may want to confirm that those cells express the expected targets, but that is outside the scope of this notebook. 
+
+## Validation of cell type assignments
+
+In the following section we'll show some plots to help validate that we are content with our cell type assignments. 
+These plots will look specifically at expression of a set of key marker genes that we expect to up in the assigned cell types. 
+These markers are found in `references/combined-markers.tsv` and include a smaller set of markers for each of the cell types we have identified. 
+
+```{r}
+# read in file with validation markers 
+validation_markers_df <- readr::read_tsv(validation_markers_file)
+
+# pull out list of genes 
+genes <- validation_markers_df |>
+  dplyr::pull(ensembl_gene_id)
+
+# get individual gene counts for all marker genes 
+gene_cts <- logcounts(sce[genes, ]) |>
+  as.matrix() |>
+  t() |> 
+  as.data.frame()
+gene_cts$barcodes <- rownames(gene_cts)
+
+# get all unique cell types from the final lumped group 
+celltypes_df <- all_info_df |> 
+  dplyr::select(barcodes, final_lumped)
+
+# create a df that has gene expression column and column indicating whether or not the gene is detected in that cell 
+genes_df <- gene_cts |> 
+  tidyr::pivot_longer(!barcodes, names_to = "ensembl_gene_id", values_to = "gene_exp") |> 
+  dplyr::left_join(celltypes_df, by = c("barcodes")) |> 
+  dplyr::left_join(validation_markers_df, by = c("ensembl_gene_id")) |> 
+  dplyr::rowwise() |> 
+  dplyr::mutate(
+    detected = gene_exp > 0 # column indicating if gene is present or not
+  )
+
+# get total number of cells per final annotation group 
+total_cells_df <- genes_df |> 
+  dplyr::select(barcodes, final_lumped) |> 
+  unique() |> 
+  dplyr::count(final_lumped, name = "total_cells")
```

Comment:
Again, I don't think you need this intermediate to join in but can just tack in an `add_counts()` into your piping below where you define `gene_summary_df`. Using `add_count()` in place of the `left_join()` on line 561 for `total_cells_df` I think will get you there